modules/m_kline.c

```diff
@@ -424,19 +430,22 @@ mo_unkline(struct MsgBuf *msgbuf_p, struct Client *client_p, struct Client *sour
 		return;
 	}
 
-	if(aconf->lifetime)
+	do
```

Comment:
This sequence kills the server:

```
KLINE 1000 10.0.0.1 :a global ban
UNKLINE 10.0.0.1 ON this.server
```

because there is nothing to end the while loop. That same global kline keeps coming up and never getting removed.